src/rust/engine/fs/src/store.rs

```diff
@@ -1019,6 +1052,48 @@ mod local {
     pub fn all_lmdbs(&self) -> Vec<(Arc<Environment>, Database, Database)> {
       self.lmdbs.values().cloned().collect()
     }
+
+    pub fn compact(&self) -> Result<(), String> {
+      for b in 0x00..0x10 {
+        let key = b << 4;
+
+        let mut dirname = {
+          let mut s = String::new();
+          fmt::Write::write_fmt(&mut s, format_args!("{:x}", key)).unwrap();
+          s[0..1].to_owned()
+        };
+        let old_dir = self.root_path.join(&dirname);
+        dirname.push_str(".tmp");
+        let new_dir = self.root_path.join(&dirname);
+        super::super::safe_create_dir_all(&new_dir).map_err(|err| {
+          format!(
+            "Error making directory for store at {:?}: {:?}",
+            new_dir, err
+          )
+        })?;
+
+        let env = Environment::new()
+          .set_flags(EnvironmentFlags::NO_SYNC | EnvironmentFlags::NO_TLS)
```

Comment:
Should extract a constant for the union of these flags.

Reply:
Pulled out fn